### PR TITLE
WIP:  Read Polhemus Montage

### PR DESCRIPTION
**Pull request by christianbrodbeck**
Read Montage from Polhemus Point files (following up on https://github.com/mne-tools/mne-python/issues/1703#issuecomment-67361977)

To do:
- [x] allow point names to be specified in a file
- [x] TESTS
- [x] update DOCs

Questions:
 - ~~Do you prefer leaving this in `io.meas_info` or would you rather move it to a dedicated `io.polhemus` module?~~
 - ~~Can it be a problem to have superfluous points in the montage? (Should I drop Nasion, LPA and RPA after using them for the transform or add them to the Montage?)~~
 - `Montage.__init__` does not check its inputs, is it only for internal use so we can assume they're safe?

### CODE REVIEW

File: mne/channels/layout.py [hunk 1]
```diff
@@ -924,3 +928,69 @@ def apply_montage(info, montage):
...
+        nasion = points[names_lower.index('nasion')]
+        lpa = points[names_lower.index('lpa')]
+        rpa = points[names_lower.index('rpa')]
+        neuromag_trans = get_ras_to_neuromag_trans(nasion, lpa, rpa)
+        points = apply_trans(neuromag_trans, points)
+
+    # drop points
+    index = np.array([bool(name) for name in point_names])
+    point_names = [name for name in point_names if name]
+    return Montage(points[index], point_names, kind, np.arange(len(points)))
```
Comment (teonlamont) [inline mne/channels/layout.py]: shouldn't the selection be np.where(points[index])[0] instead of np.arange(len(points))?

**Comment by teonlamont** [comment]
other than my most recent comment, LGTM.

File: mne/channels/layout.py [hunk 2]
```diff
@@ -924,3 +928,69 @@ def apply_montage(info, montage):
...
+        Name for the Montage.
+    transform : bool
+        Transform the points to MNE's coordinate system. This requires the
+        'Nasion', 'LPA' and 'RPA' points to be specified. When False, the
+        points are still transformed to the RAS coordinate system but are not
+        aligned using Nasion, LPA and RPA.
+
+    Returns
+    -------
+    montage : Montage
```
Comment (teonlamont) [inline mne/channels/layout.py]: instance of Montage

**Comment by christianbrodbeck** [comment]
After working through my data set I am actually less certain that this function is needed, sorry :)

My new concern is that Polhemus FastScan is not very useful for defining a montage (ordered set of points) on the head surface, so I ended up writing a small utility. Now when saving the points I think it's a good redundancy to also write channel names and not only coordinates. That means however that the point files don't correspond to the polhemus standard anymore, but instead AFAICT to one of the Montage formats (name plus x, y, z coordinates). I could thus use the same reader, however I would require the option to transform a montage into MNE space using nasion, LAP and RAP (to bring them into the same coordinate space as the FIFF head shape).

@dengemann if you say that
> the montage reader is about adding pre-specified standard locations

so is there a reader already for reading subject-specific electrode points/montages? Would you suggest to use a different function for reading subject-specific montages rather than using the same as for general montages? 

How does the landscape of tools look that produce subject specific montages? the name, x, y, z format seems reasonable to me and is easy to produce. Of course the `read_montage_polhemus` function proposed here might still be useful if other people use that route or have other tools that export in polhemus format.

**Comment by agramfort** [comment]
I don't get all the details but one general comment

```
$ python -c "import this" | grep "only one"
There should be one-- and preferably only one --obvious way to do it.
` ``

**Comment by christianbrodbeck** [comment]
Yes @agramfort most of the discussion here actually revolves around which one that is :)

**Comment by agramfort** [comment]
the simplest :)

```
$ python -c "import this" | grep "Simple"
Simple is better than complex.
```

:)

**Comment by christianbrodbeck** [comment]
Ok here is my suggestion: `read_montage()` already has a scale parameter. This scale parameter promises to modify the positions so that they are good for plotting. @teonlamont mentioned the standard coordinate system of MNE is the Neuromag system (according to the manual). To me these two transformations seem closely related. I would modify the scale parameter with 3 options along these lines:

 - `None`: no modification of positions
 - `"scale"` only scaling (current)
 - `"Neuromag"` Transformation to neuromag (this requires presence of nasion, lap and rap, hence `"scale"` is a fallback for when one wants to plot but has no fiducial coordinates.

**Comment by agramfort** [comment]
if you can avoid introducing a new function I am happy

**Comment by teonlamont** [comment]
hey, I am +1 for @christianbrodbeck's last suggestion for adding the argument 'neuromag' to scale transform the fiducials if needed. I think it would make everyone happy, no additional functions but added functionality ;)

**Comment by agramfort** [comment]
+1

**Comment by teonlamont** [comment]
closing for #1846

**Pull request closed without merging**